Classes/Capture/AddNoteViewController.swift

```diff
@@ -54,12 +54,114 @@ final class AddNoteViewController: UIViewController {
+        button1.frame = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        button1.bounds = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        let urlButton = UIBarButtonItem( customView: button1 )
+        let button2 = UIButton( type: .custom )
+        button2.setImage( UIImage( named: "datepick.png" ), for: .normal )
+        button2.addTarget(self, action: #selector(setDatePickerView), for: .touchUpInside)
+        button2.frame = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        button2.bounds = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        let dateButton = UIBarButtonItem( customView: button2 )
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        bar.items = [urlButton, dateButton, spacer ]
+        bar.sizeToFit()
+        return bar
+    }()
+
+    @objc func addLinkMarkup() {
+        guard self.textView.text != nil else { return } // If no selected text, ignore.  Consider popup?
+        if let range = self.textView.selectedTextRange, !range.isEmpty {
+            let selectedText = self.textView.text( in: range )
+            self.textView.replace( range, withText: "[[\(selectedText!)][]]" )
+            let newCursor = self.textView.position(from: range.end, offset: 4)!
+            self.textView.selectedTextRange = self.textView.textRange( from: newCursor, to: newCursor)
+        }
+    }
+
+    let datePicker: UIDatePicker = {
+        let picker = UIDatePicker(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 216))
+        picker.datePickerMode = .dateAndTime
+        return picker
+    }()
+
+    private lazy var datePickBar: UIToolbar = {
+        let bar = UIToolbar()
+        let cancelButton = UIBarButtonItem( title: "Cancel", style: .plain, target: self, action: #selector(setDefaultInputView) )
+        let scheduleButton = UIBarButtonItem( title: "Schd", style: .plain, target: self, action: #selector(insertDateSchedule) )
+        let deadlineButton = UIBarButtonItem( title: "Dead", style: .plain, target: self, action: #selector(insertDateDeadline) )
+        let agendaButton = UIBarButtonItem( title: "Agenda", style: .plain, target: self, action: #selector(insertDateAgenda) )
+        let plainButton = UIBarButtonItem( title: "Plain", style: .plain, target: self, action: #selector(insertDatePlain) )
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        bar.items = [ scheduleButton, deadlineButton, agendaButton, plainButton, spacer, cancelButton ]
+        bar.sizeToFit()
+        return bar
+    }()
+
+    private func datePicked() -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "YYYY-MM-dd E hh:mm"
+        //dateFormatter.formatOptions = [.withTime]
+        return dateFormatter.string( from: self.datePicker.date )
+    }
+
+    private func replaceSelected( with newString: String ) {
+        print( "replaceSelected" )
```

Comment:
Let's remove this as well. It is for debug purposes as I can see, but we do not need in the final version.

Reply:
Oh, wow.  I thought I removed that.  Thanks.

Reply:
done